### DISCUSSION TITLE
Is it possible to enable and disable widgets?

### DISCUSSION THREAD

**Discussion by mark-summerfield** [fltk-rs/fltk-rs#1104] [Q&A]
I have a dialog in which a button only makes sense in certain contexts. So I need to disable the button by default and only enable it when it makes sense. How can I do this? It it even possible? I tried `deactivate()` but that started the button off disabled but didn't stop it being clicked and then behaving normally. I want to be able to disable a button (and other widgets) so it won't interact at all; and then later enable it again.

**Comment by mark-summerfield** [comment]
I think the solution is to set the trigger to Never.

**Comment by KillTheMule** [comment]
I have buttons like this in my app, and running `deactivate()` on them does exactly what it says for me. Maybe you're running that at the wrong point in your code?

**Reply by mark-summerfield** [reply]
You are quite right! That works great. Thank you.
